dlt/common/libs/pyiceberg.py

```diff
@@ -20,6 +21,12 @@
 
 from dlt.destinations.impl.filesystem.filesystem import FilesystemClient
 
+if sys.version_info[:3] == (3, 9, 7):
```

Comment:
Imho the uv or poetry dependency system should warn you if you try to install pyiceberg on 3.9.7, even if it is a transitive dependency, so I don't think this is needed.